rp2040-hal/src/clocks/macros.rs

```diff
@@ -107,17 +166,31 @@ macro_rules! clock {
                         clock_nr: pac::clocks::clk_ref_ctrl::SRC_A::CLKSRC_CLK_REF_AUX as u8,
                     }
                 }
+            }
+
+            impl Clock for $name {
+                type Variant = [<$reg:camel SrcType>];
+
+                #[doc = "Get operating frequency for `"$name"`"]
+                fn freq(&self) -> Hertz {
+                    self.frequency
+                }
 
-                /// Configure this clock based on a clock source and desired frequency
-                pub fn configure_clock<S:ClockSource + ValidSrc<$name, Variant=[<$reg:camel SrcType>]>>(&mut self, src: &S, freq: Hertz) -> bool{
```

Comment:
Not that you are changing this but this should probably return a `Result<(),ClockError>` or something. Magic return values aren't really that idiomatic in Rust.

Reply:
Done